File: SuperAdmin/backend/SuperAdmin.API/Models/StoreDocument.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SuperAdmin.API.Models;

[Table("store_documents")]
public class StoreDocument
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; }

    [Column("tenant_id")]
    public Guid TenantId { get; set; }

    [Column("document_type")]
    public string DocumentType { get; set; } = string.Empty;

    [Column("file_url")]
    public string FileUrl { get; set; } = string.Empty;

    [Column("file_name")]
    public string? FileName { get; set; }

    [Column("status")]
    public string Status { get; set; } = "PENDING";

    [Column("review_notes")]
    public string? ReviewNotes { get; set; }

    [Column("uploaded_at")]
    public DateTime? UploadedAt { get; set; }

    [Column("reviewed_at")]
    public DateTime? ReviewedAt { get; set; }

    [ForeignKey("TenantId")]
    public Tenant? Tenant { get; set; }
}

File: Backend/backend/src/main/java/com/example/backend/entity/StoreDocument.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.Instant;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "store_documents")
public class StoreDocument {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tenant_id", nullable = false)
    private Tenant tenant;

    @Enumerated(EnumType.STRING)
    @Column(name = "document_type", nullable = false, length = 30)
    private DocumentType documentType;

    @Column(nullable = false)
    private String fileUrl;

    private String fileName;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private DocumentStatus status = DocumentStatus.PENDING;

    @Column(columnDefinition = "TEXT")
    private String reviewNotes;

    @CreationTimestamp
    private Instant uploadedAt;

    private Instant reviewedAt;

    public enum DocumentType {
        CIPC,
        COA,
        BANK_DETAILS,
        STOREFRONT_PHOTO
    }

    public enum DocumentStatus {
        PENDING, ACCEPTED, REJECTED
    }
}
